cascade/cascade.py

```diff
@@ -414,6 +416,29 @@ def _check_pull_output_overload(self, stderr: str) -> bool:
         """
         return any([x in stderr for x in _DOCKER_PULL_ERRORS])
 
+    def get_singularity_credentials(self, image: str) -> tuple:
```

Comment:
you may want to consider doing this once and caching it in a map so you can just look it up (as you may need to do it many times).